src/Azure/AzureAD/Authentication.AzureAD.UI/src/AzureADOptions.cs

```diff
@@ -30,30 +30,62 @@ public class AzureADOptions
         /// </summary>
         public string ClientId { get; set; }
 
         /// <summary>
-        /// Gets or sets the client secret.
+        /// Gets or sets the client secret for the application (application password) 
         /// </summary>
+        /// <remarks>
+        /// The client secret is only used if the Web app or Web API
+        /// calls a Web API
+        /// </remarks>
         public string ClientSecret { get; set; }
 
         /// <summary>
-        /// Gets or sets the tenant Id.
+        /// Gets or sets the tenant id. The tenant id can have one of the following values:
+        /// <list type="table">
+        /// <item><term>a proper tenant ID</term><description>A GUID representing the ID of the Azure Active Directory Tenant (directory ID)</description></item>
+        /// <item><term>a domain name</term><description>associated with the Azure Active Directory tenant</description></item>
+        /// <item><term>common</term><description>if the <see cref="Authority"/> is Azure AD v2.0, enables to sign-in users from any
+        /// Work and School account or Microsoft Personal Account. If Authority is Azure AD v1.0, enables sign-in from any Work and School accounts</description></item>
+        /// <item><term>organizations</term><description>if the <see cref="Authority"/> is Azure AD v2.0, enables to sign-in users from any
+        /// Work and School account</description></item>
+        /// <item><term>consumers</term><description>if the <see cref="Authority"/> is Azure AD v2.0, enables to sign-in users from any
+        /// Microsoft personal account</description></item>
+        /// </list>
         /// </summary>
         public string TenantId { get; set; }
 
         /// <summary>
         /// Gets or sets the Azure Active Directory instance.
+        /// Typical values are:
+        /// <list type="table">
+        /// <item><term>https://login.microsoftonline.com/</term><description>For Microsoft Azure public cloud</description></item>
+        /// <item><term>https://login.microsoftonline.us/</term><description>For Azure US Government</description></item>
+        /// <item><term>https://login.partner.microsoftonline.cn/</term><description>For Azure China 21Vianet</description></item>
+        /// <item><term>https://login.microsoftonline.de/</term><description>For Azure Germany</description></item>
+        /// </list>
         /// </summary>
-        public string Instance { get; set; }
+        public string Instance { get; set; } = "https://login.microsoftonline.com/";
 
         /// <summary>
-        /// Gets or sets the domain of the Azure Active Directory tennant.
+        /// Gets or sets the domain associated with the Azure Active Directory tenant.
         /// </summary>
         public string Domain { get; set; }
 
+        /// <summary>
+        /// Azure Active Directory Authority
+        /// </summary>
+        public string Authority { get; set; } = "{Instance}{TenantId}";
```

Comment:
These seem to be more like string templates and not default values. It should then be the responsability of *Configuratgion.cs to create default values if the user hasn't set any. 

i.e. Just tell the user that the default value will be {Instance}{TenantId} if he does not set anything.

Reply:
well this is {Instance}{TenantId}/v2.0 in the case of v2.0.
See also the templates.